automata/base/automaton.py

```diff
@@ -151,6 +151,10 @@ def __repr__(self) -> str:
         )
         return f"{self.__class__.__qualname__}({values})"
 
-    def __contains__(self, input_str: str) -> bool:
+    def __contains__(self, item: Any) -> bool:
```

Comment:
Changed this because mypy was complaining. Like some of the other dunder methods, I think this has to accept anything as a possible argument.

Reply:
@eliotwrobson That's fine! But if we're gonna do that, I wonder if we should have some sort of type check here to ensure that the item is an iterable or something. Or maybe it's not even worth the trouble.

Reply:
Oh in the changes I made to this function, I added an `isinstance` check that returns false if anything other than a string is passed in.

Reply:
@eliotwrobson Oh perfect, I see that now. Excellent, thank you!